code/modules/mob/living/silicon/pai/software.dm

```diff
@@ -130,8 +130,8 @@
 				if(available_software.Find(target))

```

Comment:
Instead of flipping the add and cost subtraction, add an && !(target in software) to this if